### ISSUE TITLE
Index: Script File is not running

### ISSUE THREAD

**Issue by dpieski** [simon987/sist2#86]
sist2 version: 2.7.4

Platform (Linux or Docker): Docker

Elasticsearch version: 7.7.0

Command with arguments:  `docker run -it --name sindex-theeag2 --network host -v $PWD/out/:/out simon987/sist2:2.7.4 index -t 4 --very-verbose  --script-file /out/theeagle0729/scripts.java --force-reset --batch-size 1000 --es-url http://localhost:9201 /out/theeagle0729`

Returns:

``` powershell
[7F960D339840] [2020-07-29 20:21:13] [DEBUG cli.c] arg es_url=http://localhost:9201
[7F960D339840] [2020-07-29 20:21:13] [DEBUG cli.c] arg index_path=/out/theeagle0729
[7F960D339840] [2020-07-29 20:21:13] [DEBUG cli.c] arg script_path=/out/theeagle0729/scripts.java
[7F960D339840] [2020-07-29 20:21:13] [DEBUG cli.c] arg script=ArrayList tags = ctx._source.tag = new ArrayList();
...
** Prints out the whole scripts.java file **
...
[7F960D339840] [2020-07-29 20:21:13] [DEBUG cli.c] arg print=0
[7F960D339840] [2020-07-29 20:21:13] [DEBUG cli.c] arg batch_size=1000
[7F960D339840] [2020-07-29 20:21:13] [DEBUG cli.c] arg force_reset=1
[7F960D339840] [2020-07-29 20:21:13] [INFO elastic.c] Delete index <200>
[7F960D339840] [2020-07-29 20:21:13] [INFO elastic.c] Create index <200>
[7F960D339840] [2020-07-29 20:21:13] [INFO elastic.c] Close index <200>
[7F960D339840] [2020-07-29 20:21:13] [INFO elastic.c] Create pipeline <200>
[7F960D339840] [2020-07-29 20:21:13] [INFO elastic.c] Update settings <200>
[7F960D339840] [2020-07-29 20:21:13] [INFO elastic.c] Update mappings <200>
[7F960D339840] [2020-07-29 20:21:13] [INFO elastic.c] Open index <200>
[7F960D339840] [2020-07-29 20:21:13] [DEBUG main.c] descriptor version 2.7.4 (binary)
>>[7F960D339840] [2020-07-29 20:21:13] [DEBUG store.c] Read tags for 0 documents
[7F960D339840] [2020-07-29 20:21:13] [INFO tpool.c] Starting thread pool with 4 threads
[7F960D186700] [2020-07-29 20:21:17] [INFO elastic.c] Indexed 1000 documents (4603kB) <200>
...
>>[7F960D339840] [2020-07-29 20:21:43] [INFO tpool.c] Worker threads finished
[7F960D339840] [2020-07-29 20:21:47] [INFO elastic.c] Refresh index <200>
>>[7F960D339840] [2020-07-29 20:21:47] [INFO elastic.c] Executed user script <200>
[7F960D339840] [2020-07-29 20:21:47] [INFO elastic.c] Refresh index <200>
[7F960D339840] [2020-07-29 20:21:47] [INFO elastic.c] Merge index <200>
[7F960D339840] [2020-07-29 20:21:47] [INFO elastic.c] Set refresh interval <200>
[7F960D339840] [2020-07-29 20:21:47] [INFO tpool.c] Destroying thread pool
[7F9607FFF700] [2020-07-29 20:21:47] [INFO tpool.c] Executing cleanup function
[7F9607FFF700] [2020-07-29 20:21:47] [WARNING elastic.c] calling _elastic_flush with 0 in queue
[7F9606FFD700] [2020-07-29 20:21:47] [INFO tpool.c] Executing cleanup function
[7F9606FFD700] [2020-07-29 20:21:47] [WARNING elastic.c] calling _elastic_flush with 0 in queue
[7F96077FE700] [2020-07-29 20:21:47] [INFO tpool.c] Executing cleanup function
>>[7F96077FE700] [2020-07-29 20:21:50] [INFO elastic.c] Indexed 893 documents (9871kB) <200>
[7F960D186700] [2020-07-29 20:21:50] [INFO tpool.c] Executing cleanup function
[7F960D186700] [2020-07-29 20:21:50] [WARNING elastic.c] calling _elastic_flush with 0 in queue
[7F960D339840] [2020-07-29 20:21:50] [INFO tpool.c] Final cleanup

PS C:\>
```

I put '>>' in front of a few lines I have questions about.

First, it doesn't seem to actually be executing the user script. I do not get any errors or any tags. I have intentionally added an error to the scripts.java file and nothing changes.

Second, it seem that the last group of files are not indexed until after the index is merged, during cleanup. Unless that is just a late log. 

Third, what is the "Read tags for 0 documents" comment regarding?

**Comment by simon987** [comment]
> it doesn't seem to actually be executing the user script

Thanks for the bug report, I'll try to figure out what's causing this

> it seem that the last group of files are not indexed until after the index is merged

You're right, I don't think it matters that much for now but I'll create [another issue](87) for that

> what is the "Read tags for 0 documents" comment regarding?

Before indexing, all manually added tags are read from the /tags/ database and are applied to documents before they are sent to elasticsearch. This is only useful if you need to copy tags from one index to another (and that the file paths are identical)

**Comment by simon987** [comment]
Hi  @dpieski ,
The user script is in fact running but it is running asynchronously (See [Elasticsearch task management](https://www.elastic.co/guide/en/elasticsearch/reference/current/tasks.html)) I removed that feature by default and added the `--async-script` option to enable because it was confusing. When indexing with `--async-script` there is now a log message with an url you can use to check the status/progress of the script, example:

```
./sist2 exec-script idx --script-file my_script.txt --async-script
[7F8FEAC2ECC0] [2020-08-03 19:47:44] [INFO elastic.c] User script queued: http://localhost:9200/_tasks/CMNJJj7cRzu_HEE0bWiAoQ:24781
```

Changes for this and the other issue are available in the 2.8.0 version.

**Comment by dpieski** [comment]
Awesome! Seems to be working great! (Although I have not tried out the `--async-script`)
Thanks
